corelib/src/array.cairo

```diff
@@ -121,14 +121,7 @@ impl ArrayTCloneImpl<T, impl TClone: Clone::<T>, impl TDrop: Drop::<T>> of Clone
 fn clone_loop<T, impl TClone: Clone::<T>, impl TDrop: Drop::<T>>(
     mut span: Span<T>, ref response: Array<T>
 ) {
-    match withdraw_gas() {
-        Option::Some(_) => {},
-        Option::None(_) => {
-            let mut data = array_new();
-            array_append(ref data, 'Out of gas');
-            panic(data);
-        },
-    }
+    withdraw_gas().expect('Out of gas');
```

Comment:
It appears this change causes `Failed calculating gas usage, it is likely a call for `gas::withdraw_gas` is missing.`

```rust
match withdraw_gas() {
    Option::Some(_) => {},
    Option::None(_) => panic_with_felt252('Out of gas'),
}
```
works though